lib/bundler/shared_helpers.rb

```diff
@@ -19,6 +19,16 @@ def requirement
 
 module Bundler
   module SharedHelpers
+    def default_root
+      given_gemfile = ENV["BUNDLE_GEMFILE"]
+      return Pathname.new(given_gemfile).dirname if given_gemfile && !given_gemfile.empty?
+      search_up("") do |path|
```

Comment:
this still can return `~`

Reply:
I have rebased the thing and it looks like my changes here weren't necessary anymore, so I have removed them.